### DISCUSSION TITLE
BRouter download manager in Cruiser

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#161] [Q&A] [answered]
Use BRouter download manager inside Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/b0e6a1f0-0add-467f-b43e-9ea54215ed7c" width="50%"/>

**Comment by 0709wiwiwi** [comment]
Tools ?
In tools, we find the download for the GH routing graphs.
The Brouter counterpart are then the rd5 Brouter files.
But this concerns the downloads of the brf routing profiles ? 
Better in Settings anyway ? Other opinions are needed.

**Comment by DolomitesTom** [comment]
independent from menu, very good idea to download brouter files within Cruiser.

**Comment by 0709wiwiwi** [comment]
@ Emux.
I suppose Android does not allow to interact directly with the Brouter folder ?
So can you reveal a little where you will place the Cruiser "profiles2" folder ?
Will the "profiles2" folder present in the Brouter app then be totally ignored ?
Benefit: Clarity.

Compare with:
Locus map **Classic** uses A + B 
A. Default: Internal BRouter folder "profiles2"
B. Custom: Locus folder "router" subfolder "profiles 2".
More confusing, please not.

**Reply by devemux86** [reply]
@0709wiwiwi

I'll be able to provide more details once there's more progress into this feature. :slightly_smiling_face: 
I still need to check some things.

**Reply by 0709wiwiwi** [reply]
> @ Emux.
> I suppose Android does not allow to interact directly with the Brouter folder ?

_Download then will have to come from which source ? Brouter web ? 
(Many profiles available , fairly few are really interesting or useful)
Find the "profiles2" folder in the (accessible) Brouter Android media.  
Auto transfer (copy) of this folder into Cruiser is possible ? Y/N ?
Advantage: No extra user action so is necessary. KISS._

**Reply by devemux86** [reply]
@0709wiwiwi

App folders are only accessible from file manager apps or Android file browser (Cruiser open / save dialogs).
Apps cannot process directly other app folders.

I will see how we can improve the whole implementation.
Everyone's ideas are helpful! :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
- So the Cruiser app can't ask the **default** Android file explorer (workhorse) to copy the Brouter "profiles2" folder to Cruiser android media.
- However, users can use that same **default** Android file explorer to simply copy the Brouter "profiles2" folder into Cruiser android media. Tested with A12 phone, this runs like a piece of cake.

Sorry I'm not urging you to be tricky. 
But because I'm not entirely sure what apps still can or can't do. 
Sure previous Android versions were definitely much friendlier.

**Reply by devemux86** [reply]
If I implement the idea described below, there is no need for manual copy.

Cruiser can include the BRouter profiles and could be in the /media/ folder so users can still edit them.

- https://github.com/devemux86/cruiser/discussions/56#discussioncomment-5839071

**Reply by 0709wiwiwi** [reply]
Probably the most friendly and yet still sufficiently flexible solution.

Note.
In the meantime, I have already downgraded back to version 3.018.
(Prepare & execute a downgrade I am starting to get the hang of).

**Comment by airald** [comment]
IMO should be in Settings.

**Comment by DolomitesTom** [comment]
In my opinion, it belongs better in the menu map. I found POIs download in the Tools menu a bit hard to get used to.

**Comment by devemux86** [comment]
We should organize the download managers better, probably together somewhere.

**Comment by devemux86** [comment]
All download managers will be together on one screen in the next Cruiser.

Please see the updated image in the first post:
- https://github.com/devemux86/cruiser/discussions/161#discussion-5185728

**Comment by devemux86** [comment, marked as answer]
BRouter download manager is now available inside [Cruiser 3.0.20](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Menu "Map | Downloads | BRouter".
